frontend/public/components/marketplace/kubernetes-marketplace-items.jsx

```diff
@@ -151,62 +165,145 @@ export class MarketplaceTileViewPage extends React.Component {
     } else {
-      filters[filterType][id].active = value;
+      const groupFilter = _.cloneDeep(activeFilters[filterType]);
+      _.set(groupFilter, [id, 'active'], value);
+      CategoryFilterUtils.updateURLParams(filterType, CategoryFilterUtils.getFilterSearchParam(groupFilter));
     }
-    this.setState({filters});
+
+    const updatedFilters = CategoryFilterUtils.updateActiveFilters(activeFilters, filterType, id, value);
+
+    this.updateMountedState(this.getUpdatedState(categories, selectedCategoryId, updatedFilters));
   }
 
-  renderTiles() {
-    const items = this.state.filteredItems;
-    const { openOverlay } = this.props;
+  onKeywordChange(value) {
+    this.onFilterChange('keyword', null, value);
+  }
 
+  openOverlay(detailsItem) {
+    CategoryFilterUtils.updateURLParams (CategoryFilterUtils.DETAILS_ITEM_URL_PARAM, detailsItem.uid);
+    this.updateMountedState({ detailsItem });
+  }
+
+  closeOverlay() {
+    CategoryFilterUtils.updateURLParams (CategoryFilterUtils.DETAILS_ITEM_URL_PARAM, '');
+    this.updateMountedState({ detailsItem: null });
+  }
+
+  renderTabs(category, selectedCategoryId) {
+    const { id, label, subcategories, numItems } = category;
+    const active = id === selectedCategoryId;
+    const shown = id === 'all';
+
+    const tabClasses = classNames('text-capitalize', { 'co-catalog-tab__empty': !numItems });
+    return <VerticalTabs.Tab
+      key={id}
+      title={label}
+      active={active}
+      className={tabClasses}
+      onActivate={() => this.selectCategory(id)}
+      hasActiveDescendant={CategoryFilterUtils.hasActiveDescendant(selectedCategoryId, category)}
+      shown={shown}>
+      {!_.isEmpty(subcategories) && <VerticalTabs restrictTabs activeTab={CategoryFilterUtils.isActiveTab(selectedCategoryId, category)}>
+        {_.map(subcategories, subcategory => this.renderTabs(subcategory, selectedCategoryId))}
+      </VerticalTabs>}
+    </VerticalTabs.Tab>;
+  }
+
+  renderCategoryTabs(selectedCategoryId) {
+    const { categories } = this.state;
+    const activeTab = !!_.find(categories, {id: selectedCategoryId});
+
+    return <VerticalTabs restrictTabs activeTab={activeTab} shown="true">
+      {_.map(categories, (category) => this.renderTabs(category, selectedCategoryId))}
+    </VerticalTabs>;
+  }
+
+  renderSidePanel() {
+    const { activeFilters, filterCounts } = this.state;
     return (
-      <CatalogTileView.Category totalItems={items.length} viewAll={true}>
-        {_.map(items, item => {
+      <FilterSidePanel>
+        <FilterSidePanel.Category key="keyword" onSubmit={(e) => e.preventDefault()}>
+          <FormControl
+            type="text"
+            inputRef={(ref) => this.filterByKeywordInput = ref}
+            autoFocus={true}
+            placeholder="Filter by keyword..."
+            bsClass="form-control"
+            value={activeFilters.keyword.value}
+            onChange={e => this.onKeywordChange(e.target.value)}
```

Comment:
Why not make `onKeywordChange` take an event and bind to this, then you can use

```suggestion
            onChange={this.onKeywordChange}
```